ts/src/base/Exchange.ts

```diff
@@ -3976,6 +3976,18 @@ export default class Exchange {
         return this.parseNumber (value, defaultNumber);
     }
 
+    safeIntegerMultiplied (container, key, multiplier: number, defaultValue = undefined) {
```

Comment:
"product" suffix there is kinda unintuitive imo

Reply:
the thing is, usually we have SafeX, safeX2. safeXN, adding only this version is not consistent, second I don't think we should have both because it will be confusing upon programming to have a prompt of two very similar methods. Third thing,  fixing the current implementation should be very easy (for instance the python version does not seem affected by this issue)

```Typescript
const safeIntegerProduct = (o: implicitReturnType, k: string | number, $factor: number, $default?: number): number => {
    const n = prop (o, k)
    return isNumber (n) ? parseInt (n * $factor as any) : $default
};
```

![image](https://github.com/ccxt/ccxt/assets/43336371/6a8377b0-7d75-4265-9492-6556c8e90ebb)

Reply:
Also, these kind of utilities methods I don't think we should transpile, because we might find a more efficient way of doing it that requires a "per-language" solution

Reply:
bcz we would have transpilable version, which is easy to maintain, than having it in all current and upcoming languages added separately. also, there is not much extra work needed, is it ?

Reply:
> might find a more efficient way of doing it that requires a "per-language" solution

I would agree on that, but specifically this method is more common (first is to extract value with `safeValue` which is same in all langs) and second, if it ever happens, only then we can dissect it from transpilable. atm, when you look at this specific method, is there any better way for language specific optimization ? bcz this method won't be amended ever again probably.

Reply:
but sure, I won't keep you on this more time, i'll just edit as you say. above was just my thought.

Reply:
bcz if we follow that logic, we might find that part of currently transpiled methods might have more performant language-specific alternative, and we would need to migrate back them from transpilable part. like, iteration of objects or etc.

however, i've pushed the update atm, plz revise now.